images/opentelemetry-collector-contrib/tests/main.tf

```diff
@@ -105,15 +120,52 @@ resource "helm_release" "open-telemetry-custom-config" {
       {
         name : "custom-vm",
         configMap : {
-          name : kubernetes_config_map.open-telemetry-custom-config.metadata[0].name
+          name : "otelc-daemonset"
+          items : [
+            {
+              key : "custom-deploy-config.yaml"
+              path : "custom-config.yaml"
+            }
+          ]
         }
       }
     ]
-  })]
+  }
 }
 
-module "custom_helm_cleanup" {
-  source    = "../../../tflib/helm-cleanup"
-  name      = helm_release.open-telemetry-custom-config.id
-  namespace = helm_release.open-telemetry-custom-config.namespace
+resource "imagetest_feature" "basic" {
+  harness     = imagetest_harness_k3s.this
+  name        = "Basic"
+  description = "Basic functionality of the otel-collector-contrib helm chart."
+
+  steps = [
+    {
+      name = "Create namespace and configmap with custom configuration for the deployment"
+      cmd  = <<EOF
+        NAMESPACE=otelc-deploy
+        kubectl create namespace $NAMESPACE
+        kubectl create configmap $NAMESPACE --namespace $NAMESPACE --from-file=/tests/custom-deploy-config.yaml
+      EOF
+    },
+    {
+      name = "Helm deploy opentelemetry-collector-contrib as a deployment"
+      cmd  = module.helm-otelc-deploy.install_cmd
+    },
+    {
+      name = "Create namespace and configmap with custom configuration for the daemonset"
+      cmd  = <<EOF
+        NAMESPACE=otelc-daemonset
+        kubectl create namespace $NAMESPACE
+        kubectl create configmap $NAMESPACE --namespace $NAMESPACE --from-file=/tests/custom-ds-config.yaml
+      EOF
+    },
+    {
+      name = "Helm deploy opentelemetry-collector-contrib as a daemonset"
+      cmd  = module.helm-otelc-daemonset.install_cmd
```

Comment:
Does this do verification on install or is there a way to verify it's working?

Reply:
Semantically this is the same test from before, just using imagetest instead.